hopsworks-api/src/main/java/io/hops/hopsworks/api/zeppelin/notebook/repo/HDFSNotebookRepo.java

```diff
@@ -357,24 +363,42 @@ public synchronized void save(Note note, AuthenticationInfo subject) throws
   @Override
   public void remove(String noteId, AuthenticationInfo subject) throws
           IOException {
-    DistributedFileSystemOps udfso = null;
+    DistributedFileSystemOps dfso = null;
+    
     try {
-      udfso = getUserDfs(subject);
-      Path rootDir = getRootDir(udfso);
+      dfso = getDfs(superuser);
+      Path rootDir = getRootDir(dfso);
+      String hdfsOwner = dfso.getFileStatus(rootDir).getOwner();
+      
+      Matcher psuMatcher = psuPattern.matcher(hdfsOwner);
+      if (psuMatcher.matches()) {
+        String extractedPSUProjectname = psuMatcher.group(1);
+        Matcher pguMatcher = pguPattern.matcher(subject.getUser());
+        if (pguMatcher.matches()) {
+          String extractedPGUProjectname = pguMatcher.group(1);
+          if (!extractedPSUProjectname.equals(extractedPGUProjectname)) {
+            throw new IOException("User <" + subject.getUser() + "> does not " +
+                "belong to project");
+          }
+        }
+      }
+      
       Path noteDir = new Path(rootDir, noteId);
   
-      if (!udfso.getFilesystem().exists(noteDir)) {
+      if (!dfso.getFilesystem().exists(noteDir)) {
         // nothing to do
         return;
       }
   
-      if (!udfso.getFilesystem().isDirectory(noteDir)) {
+      if (!dfso.getFilesystem().isDirectory(noteDir)) {
         // it does not look like zeppelin note savings
         throw new IOException("Can not remove " + noteDir.toString());
       }
-      udfso.getFilesystem().delete(noteDir, true);
+      dfso.getFilesystem().delete(noteDir, true);
     } finally {
-      closeDfsClient(udfso);
+      if (dfso != null) {
```

Comment:
@kouzant Just to be more coherent, although for dfso it makes no difference, wouldn't it be better to still use closeDfsClient()?

Reply:
At the time there was a bug, fixed by https://github.com/hopshadoop/hopsworks/pull/762
Updated the code also here.